### PR TITLE
Isolate Schema Sharding/MERGE tests into a new file

### PR DESCRIPTION
**Pull request by tejeswarm**
Tests MERGE with combination of Schema-Sharded tables and row-sharded tables.

### CODE REVIEW

**Review by onderkalaci — APPROVED** [review]
looks good

File: src/backend/distributed/planner/merge_planner.c [hunk 1]
```diff
@@ -1185,8 +1185,8 @@ SourceResultPartitionColumnIndex(Query *mergeQuery, List *sourceTargetList,
 	if (IsCitusTableType(targetRelation->relationId, SINGLE_SHARD_DISTRIBUTED))
 	{
 		ereport(ERROR, (errmsg("MERGE operation on non-colocated "
-							   "distributed table(s) without a shard "
-							   "key is  not yet supported")));
+							   "schema sharded table(s) "
```
Comment (mulander) [inline src/backend/distributed/planner/merge_planner.c:1188]: We started referring to it as "tables/table in a distributed schema" in other messages/warnings/errors.

How about: "MERGE operation on tables in a distributed schema is not supported".  That however feels outside of what I expected (MERGE working within a single schema). Is this the case where the user tries to do a MERGE with tables across two distributed schemas (or a distributed schema and a distributed table that is not co-located)? In that case I would reword it to:

"MERGE operation across distributed schemas or with a non-colocated distributed table is not supported"

Comment (tejeswarm) [reply src/backend/distributed/planner/merge_planner.c:1188]: I will change it to "MERGE operation across distributed schema is not supported"
